engine/src/main/java/org/terasology/identity/ClientIdentity.java

```diff
@@ -49,4 +50,21 @@ public PrivateIdentityCertificate getPlayerPrivateCertificate() {
         }
         return playerPrivateCertificate;
     }
+
+    @Override
+    public int hashCode() {
+        return playerPrivateCertificate.hashCode() ^ playerPrivateCertificate.hashCode();
```

Comment:
I think that this will always be zero. Maybe just just `31 * playerPublicCertificate` instead?